crates/subspace-networking/src/create.rs

```diff
@@ -243,20 +199,20 @@ pub async fn create(
         let (pieces_by_range_request_handler, pieces_by_range_protocol_config) =
             PiecesByRangeRequestHandler::new(pieces_by_range_request_handler);
 
-        let behaviour = Behavior::new(
-            BehaviorConfig {
-                peer_id: local_peer_id,
-                bootstrap_nodes,
-                identify,
-                kademlia,
-                gossipsub,
-                value_getter,
-                pieces_by_range_protocol_config,
-                pieces_by_range_request_handler: Box::new(pieces_by_range_request_handler),
-                relay_config: relay_config_for_swarm.clone(),
-            },
+        let is_relay_server = !listen_on.is_empty() && relay_server_address.is_none();
```

Comment:
Why `!listen_on.is_empty() ` is a part of the expression? Isn't `relay_server_address` is sufficient indicator for it? If both are required what will happen when `relay_server_address.is_none() == true` and `!listen_on.is_empty() == false` (should be an error then)?

Reply:
> Node is useless as a relay server if it doesn't listen on any interface, isn't it?
Exactly what I meant. Probably, to return an error could a good idea in this case.

Reply:
This is just a peer that doesn't run a relay server and doesn't need relay server to be reachable, it has its own public interface.

Reply:
Hmm, I thought we narrowed our use-case down to very specific relay client and server cases. OK.